### PR TITLE
AG-10518 Create new donut series type

### PR DESCRIPTION
**Pull request by jacobp100**
https://ag-grid.atlassian.net/browse/AG-10518

### CODE REVIEW

File: packages/ag-charts-community/src/chart/series/polar/__image_snapshots__/donut-series-test-ts-donut-series-create-one-zerosum-donut-1-snap.png
Comment (jacobp100) [inline packages/ag-charts-community/src/chart/series/polar/__image_snapshots__/donut-series-test-ts-donut-series-create-one-zerosum-donut-1-snap.png:1]: Apparently this is expected?

File: packages/ag-charts-community/src/chart/series/polar/donutSeriesModule.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+export const DonutSeriesModule: SeriesModule<'donut'> = {
+    type: 'series',
+    optionsKey: 'series[]',
+    packageType: 'community',
+    chartTypes: ['polar'],
+
+    identifier: 'donut',
+    instanceConstructor: DonutSeries,
+    seriesDefaults: {},
+    themeTemplate: {
```
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/series/polar/donutSeriesModule.ts]: Nit: Could we trivially share this config with `pieSeriesModule.ts`?

File: packages/ag-charts-community/src/chart/series/polar/donutSeriesModule.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        innerLabels: {
+            fontStyle: undefined,
+            fontWeight: undefined,
+            fontSize: 12,
+            fontFamily: DEFAULT_FONT_FAMILY,
+            color: DEFAULT_LABEL_COLOUR,
+            margin: 2,
+        },
+    },
+    paletteFactory: ({ takeColors, colorsCount, userPalette }) => {
```
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/series/polar/donutSeriesModule.ts]: Ditto

File: packages/ag-charts-community/src/chart/series/polar/pieSeriesProperties.ts [hunk 3]
```diff
@@ -33,24 +35,34 @@ export class PieTitle extends Caption {
     showInLegend = false;
 }
 
-export class DoughnutInnerLabel<T extends object = any> extends Label<AgPieSeriesLabelFormatterParams> {
-    @Validate(STRING)
-    text!: string;
+/** @deprecated Remove this */
```
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/series/polar/pieSeriesProperties.ts]: Nit: `@deprecated` is somewhat redundant in implementation code, typically we search for `@Deprecated()` properties to find these; also typically the deprecation warnings in our own code aren't really that useful.

File: packages/ag-charts-community/src/options/series/polar/pieOptions.ts [hunk 4]
```diff
@@ -89,13 +74,22 @@ export interface AgPieSeriesThemeableOptions<TDatum = any> extends AgBaseSeriesT
     strokeWidth?: PixelSize;
     /** The rotation of the pie series in degrees. */
     rotation?: Degree;
-    /** The offset in pixels of the outer radius of the series. Used to construct doughnut charts. */
+    /**
+     * @deprecated v9.0.2 Use a Donut Series instead.
```
Comment (alantreadway) [inline packages/ag-charts-community/src/options/series/polar/pieOptions.ts]: Nit: We should use `v9.1.0` as the version number in these deprecation messages, as that will be the first release with these changes.

File: packages/ag-charts-website/src/content/docs/animation/_examples/duration/index.html [hunk 5]
```diff
@@ -3,7 +3,7 @@
         <button onclick="changeSeriesBar()">Bar</button>
         <button onclick="changeSeriesLine()">Line</button>
         <button onclick="changeSeriesArea()">Area</button>
-        <button onclick="changeSeriesPie()">Pie</button>
```
Comment (alantreadway) [inline packages/ag-charts-website/src/content/docs/animation/_examples/duration/index.html:6]: I wonder if we should keep `pie` for one of these animation examples, for the coverage?

File: packages/ag-charts-community/src/chart/series/polar/donutSeries.ts [hunk 6]
```diff
@@ -0,0 +1,1482 @@
...
+            { x: box.x + dx, y: box.y + dy },
+            { x: box.x + box.width + dx, y: box.y + dy },
+            { x: box.x + box.width + dx, y: box.y + box.height + dy },
+            { x: box.x + dx, y: box.y + box.height + dy },
+        ];
+        const sur2 = surroundingRadius ** 2;
+        return corners.some((corner) => corner.x ** 2 + corner.y ** 2 > sur2);
+    }
+
+    private computeCalloutLabelCollisionOffsets() {
```
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/series/polar/donutSeries.ts:854]: Could we factor out common functions (into `pieUtil.ts`?) for these label collision functions?

File: packages/ag-charts-community/src/chart/series/polar/donutSeries.ts [hunk 7]
```diff
@@ -0,0 +1,1482 @@
...
+    }
+
+    updateCalloutLineNodes() {
+        const { calloutLine } = this.properties;
+        const calloutLength = calloutLine.length;
+        const calloutStrokeWidth = calloutLine.strokeWidth;
+        const calloutColors = calloutLine.colors ?? this.properties.strokes;
+        const { offset } = this.properties.calloutLabel;
+
+        this.calloutLabelSelection.selectByTag<Line>(PieNodeTag.Callout).forEach((line, index) => {
```
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/series/polar/donutSeries.ts:768]: Could we factor out a common update function here?

File: packages/ag-charts-community/src/chart/series/polar/donutSeries.ts
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/series/polar/donutSeries.ts:1]: I've made some specific comments in this file, but generally could we factor out any common functions (`Selection.forEach()` callbacks seem low-hanging fruit)?

File: packages/ag-charts-website/src/content/docs/animation/_examples/duration/index.html (hunk 5 above)
Comment (jacobp100) [reply packages/ag-charts-website/src/content/docs/animation/_examples/duration/index.html:6]: As in add a new pie chart to this example? They're both donuts at the moment

Comment (jacobp100) [reply packages/ag-charts-website/src/content/docs/animation/_examples/duration/index.html:6]: Added a pie series

File: packages/ag-charts-community/src/chart/series/polar/donutSeries.ts (hunk 6 above)
Comment (jacobp100) [reply packages/ag-charts-community/src/chart/series/polar/donutSeries.ts:854]: Will revisit later

File: packages/ag-charts-community/src/chart/series/polar/donutSeries.ts (hunk 7 above)
Comment (jacobp100) [reply packages/ag-charts-community/src/chart/series/polar/donutSeries.ts:768]: Will revisit later

File: packages/ag-charts-community/src/chart/series/polar/donutSeries.ts
Comment (jacobp100) [reply packages/ag-charts-community/src/chart/series/polar/donutSeries.ts:1]: Will revisit later

**Review by alantreadway — APPROVED** [review]
LGTM, merge once CI is green - thanks!

**Pull request merged**